hack/e2e.sh

```diff
@@ -89,18 +99,30 @@ PE_URL=$(oc get route cincinnati-policy-engine -o jsonpath='{.spec.host}')
 export GRAPH_URL="http://${PE_URL}/api/upgrades_info/v1/graph"
 
 # Wait for route to become available
-ATTEMPTS=10
 DELAY=10
-
-while [ $ATTEMPTS -ge 0 ]; do
+for i in $(seq 10); do
   CODE=$(curl -s -o /dev/null -w "%{http_code}" --header 'Accept:application/json' "${GRAPH_URL}?channel=a")
   if [ "${CODE}" == "200" ]; then
     break
-  else
-    sleep ${DELAY}
-    ATTEMPTS=$((ATTEMPTS-1))
   fi
+  sleep ${DELAY}
+done
+
+# Wait for cincinnati metrics to be recorded
+# Find out the token Prometheus uses from its serviceaccount secrets
+# and use it to query for GB build info
+export PROM_ENDPOINT=$(oc -n openshift-monitoring get route thanos-querier -o jsonpath="{.spec.host}")
+export PROM_TOKEN=$(oc -n openshift-monitoring get secret \
+  $(oc -n openshift-monitoring get serviceaccount prometheus-k8s \
+    -o jsonpath='{range .secrets[*]}{.name}{"\n"}{end}' | grep prometheus-k8s-token) \
+  -o go-template='{{.data.token | base64decode}}')
+
+DELAY=30
+for i in $(seq 10); do
+  curl -kLs -H "Authorization: Bearer ${PROM_TOKEN}" "https://${PROM_ENDPOINT}/api/v1/query?query=cincinnati_gb_build_info" | grep "metric" && break
+
+  sleep ${DELAY}
 done
 
-# Run e2e tests
-/usr/bin/cincinnati-e2e-test
+# Run e2e tests in sequential mode to have SLO tests in the end
+env RUST_TEST_TASKS=1 /usr/bin/cincinnati-e2e-test
```

Comment:
Do you actually need them to run at the end, or just not in parallel to any others?

Reply:
We do - its pointless to collect metrics if there are more tests planned.

Hopefully this is a temporary solution - initially I tried making different binaries for that, but Rust mixes up binary names so Dockerfile.e2e becomes a mess.

Reply:
Please explain how we know that the SLO tests are at the end of the sequence now as it's not obvious to me.

Reply:
Rust sorts tests alphabetically and starts them at the same time. So in order to run SLO tests lasts in this PR:
* make sure the package is named "slo" so that it would run after "e2e"
* use RUST_TEST_TASKS=1 to run tests sequentially - meaning e2e tests would start one by one and then start SLO verification tests

Reply:
I see, thanks! Let's hope it keeps the alphabetical ordering until we figure out a more explicit way for the ordering.